src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,880 @@
+        """
+        if parameter is None:
+            parameter = self.parent.datamodel.HybridPlatform.PrimaryPumpState
+        if parameter is not self.parent.datamodel.HybridPlatform.PrimaryPumpState:
+            raise ValueError("Incorrect parameter passed to _updatePrimaryPumpOn. Parameter should be "
+                             "datamodel.HybridPlatform.PrimaryPumpState. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName != "Actual":
+            return
+        state = str(parameter.Actual).lower() == "true"
+        logging.debug("Primary pump turned %s." % ("on" if state else "off"))
+        self.primaryPumpOn._set_value(state, force_write=True)
+
+    def _updateNitrogenPressure(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads pressure on nitrogen inlet to the turbopump from the Orsay server and saves it in the nitrogenPressure VA
+        """
+        if parameter is None:
+            parameter = self._system.Manometer1.Pressure
+        if parameter is not self._system.Manometer1.Pressure:
+            raise ValueError("Incorrect parameter passed to _updateNitrogenPressure. Parameter should be "
+                             "datamodel.HybridPlatform.PumpingSystem.Manometer1.Pressure. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName == "Actual":
+            return
+        self.nitrogenPressure._set_value(float(parameter.Actual), force_write=True)
+
+    def terminate(self):
+        """
+        Called when Odemis is closed
+        """
+        self._system.Manometer1.ErrorState.Unsubscribe(self._updateErrorState)
+        self._system.TurboPump1.ErrorState.Unsubscribe(self._updateErrorState)
+        self._system.TurboPump1.Speed.Unsubscribe(self._updateSpeed)
+        self._system.TurboPump1.Temperature.Unsubscribe(self._updateTemperature)
+        self._system.TurboPump1.Power.Unsubscribe(self._updatePower)
+        self._system.TurboPump1.SpeedReached.Unsubscribe(self._updateSpeedReached)
+        self._system.TurboPump1.IsOn.Unsubscribe(self._updateTurboPumpOn)
+        self.parent.datamodel.HybridPlatform.PrimaryPumpState.Unsubscribe(self._updatePrimaryPumpOn)
+        self._system.Manometer1.Pressure.Unsubscribe(self._updateNitrogenPressure)
+        self._system = None
+
+
+class UPS(model.HwComponent):
+    """
+    This represents the uniterupted power supply from Orsay Physics
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Defines the following VA's and links them to the callbacks from the Orsay server:
+        • level (FloatContinuous, range=(0.0, 1.0), read-only, unit is "",
```

Comment:
Instead of saying "unit is """, I'd suggest to clarify that this is the ratio between empty and fully charged.